src/impl_/extract_argument.rs

```diff
@@ -268,13 +308,21 @@ impl FunctionDescription {
 
         // Handle keyword arguments
         let mut varkeywords = K::Varkeywords::default();
-        if let Some(kwnames) = py.from_borrowed_ptr_or_opt::<PyTuple>(kwnames) {
+
+        // Safety: kwnames is known to be a pointer to a tuple, or null
+        let kwnames: &'py Option<Bound<'py, PyTuple>> = std::mem::transmute(&kwnames);
```

Comment:
Maybe we want to create a `Bound::ref_from_ptr_or_opt` to avoid `std::mem::transmute`?

Reply:
In the end I adjusted the `iter_borrowed()` implementations to allow calling them from a `Borrowed` instance, and then I was able to use `Borrowed::from_ptr` methods in this code here. I think it's fine to use `Borrowed` instead of `ref_from_ptr` here because we can control that pesky borrowed lifetime `'a` without worrying about user code, unlike the macros.